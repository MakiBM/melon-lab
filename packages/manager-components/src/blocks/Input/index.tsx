import classNames from 'classnames';
import React, { ChangeEventHandler, StatelessComponent } from 'react';

import styles from './styles.css';

export interface InputProps {
  disabled?: boolean;
  placeholder?: string;
  onChange?: ChangeEventHandler<Element>;
  label: string;
  type: string;
  insideLabel?: boolean;
  pattern?: string;
  step?: string;
  cleanNumber?: boolean;
  decimals?: number;
  value?: string;
}

const Input: StatelessComponent<InputProps> = ({
  disabled,
  placeholder,
  onChange,
  label,
  type = 'text',
  insideLabel,
  step,
  pattern,
  decimals = 4,
  value,
  cleanNumber,
}) => {
  const classname = classNames('input', {
    'input--inside-label': insideLabel,
  });

  return (
    <div className={classname}>
      <style jsx>{styles}</style>
      <span className="input__label">{label}</span>
      <input
        pattern={pattern}
        step={step}
        className="input__field"
        type={type}
        disabled={disabled}
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
    </div>
  );
};

export default Input;
