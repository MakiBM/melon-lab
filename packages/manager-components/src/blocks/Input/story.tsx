import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import React from 'react';
import Input from './index';

storiesOf('Blocks|Input', module)
  .add('Default', () => {
    return <Input name="text" type="text" onChange={action('changed')} />;
  })
  .add('Disabled', () => {
    return <Input name="text" type="text" disabled={true} />;
  })
  .add('With placeholder', () => {
    return <Input name="text" type="text" placeholder="Placeholder..." />;
  })
  .add('With label', () => {
    return (
      <Input
        name="text"
        type="text"
        placeholder="Placeholder..."
        label="Label"
      />
    );
  })
  .add('With inside label', () => {
    return (
      <Input
        name="text"
        type="number"
        placeholder="0.000"
        label="Label"
        insideLabel={true}
      />
    );
  });
