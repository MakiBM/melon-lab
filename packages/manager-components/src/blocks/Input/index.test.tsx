import React from 'react';
import Input from './index';

describe('Input', () => {
  const defaultElement = <Input />;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });
});
