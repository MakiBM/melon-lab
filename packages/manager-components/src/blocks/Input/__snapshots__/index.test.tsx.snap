// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should pass a typed value to the onChange handler 1`] = `
<div
  className="jsx-__scopedHash input"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <span
    className="jsx-__scopedHash input__label"
  />
  <input
    className="jsx-__scopedHash input__field"
    onChange={[MockFunction]}
    type="text"
  />
  <div
    className="jsx-__scopedHash input__error"
  />
</div>
`;

exports[`Input should render correctly 1`] = `
<div
  className="jsx-__scopedHash input"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <span
    className="jsx-__scopedHash input__label"
  />
  <input
    className="jsx-__scopedHash input__field"
    type="text"
  />
  <div
    className="jsx-__scopedHash input__error"
  />
</div>
`;
